rhp/v4/rhp_test.go

```diff
@@ -19,3 +20,212 @@ func TestMinRenterAllowance(t *testing.T) {
+			},
+			RenterCost: prices.ContractPrice.Add(prices.StoragePrice.Mul64(SectorSize).Mul64(extension)), // storage cost is calculated for just the extension
+			HostCost:   types.ZeroCurrency,                                                               // collateral lock up is less than rollover
+		},
+		{
+			Description: "renewed storage",
+			Modify: func(rev *types.V2FileContract, p *RPCRenewContractParams) {
+				// add storage
+				rev.Capacity = SectorSize
+				rev.Filesize = SectorSize
+
+				// adjust the renewal params
+				p.Allowance = rev.RenterOutput.Value.Add(types.Siacoins(20))
+				p.Collateral = rev.TotalCollateral.Add(types.Siacoins(10))
+			},
+			RenterCost: types.Siacoins(20).Add(prices.ContractPrice).Add(prices.StoragePrice.Mul64(SectorSize).Mul64(extension)), // storage cost is calculated for just the extension
+			HostCost:   types.Siacoins(10).Add(prices.Collateral.Mul64(SectorSize).Mul64(renewalDuration)),                       // collateral is calculated for the full duration
+		},
+		{
+			Description: "renewed storage - no renter rollover",
+			Modify: func(rev *types.V2FileContract, p *RPCRenewContractParams) {
+				// adjust the renewal params
+				p.Allowance = rev.RenterOutput.Value.Add(types.Siacoins(20))
+				p.Collateral = rev.TotalCollateral.Add(types.Siacoins(10))
+
+				// add storage
+				rev.Capacity = SectorSize
+				rev.Filesize = SectorSize
+				// transfer all the renter funds to the host so the renter will need to put up more allowance
+				rev.HostOutput.Value, rev.RenterOutput.Value = rev.HostOutput.Value.Add(rev.RenterOutput.Value), types.ZeroCurrency
+			},
+			RenterCost: types.Siacoins(320).Add(prices.ContractPrice).Add(prices.StoragePrice.Mul64(SectorSize).Mul64(extension)), // storage cost is calculated for just the extension
+			HostCost:   types.Siacoins(10).Add(prices.Collateral.Mul64(SectorSize).Mul64(renewalDuration)),                        // collateral is calculated for the full duration
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.Description, func(t *testing.T) {
+			contract, _ := NewContract(prices, RPCFormContractParams{
+				RenterPublicKey: renterKey,
+				RenterAddress:   types.StandardAddress(renterKey),
+				Allowance:       types.Siacoins(300),
+				Collateral:      types.Siacoins(400),
+				ProofHeight:     initialProofHeight,
+			}, hostKey, types.StandardAddress(hostKey))
+
+			params := RPCRenewContractParams{
+				ProofHeight: renewalProofHeight,
+			}
+			tc.Modify(&contract, &params)
+
+			prices.TipHeight = renewalHeight
+			renewal, _ := RenewContract(contract, prices, params)
+			tax := cs.V2FileContractTax(renewal.NewContract)
+			renter, host := RenewalCost(cs, prices, renewal, types.ZeroCurrency)
+			if !renter.Equals(tc.RenterCost.Add(tax)) {
+				t.Errorf("expected renter cost %v, got %v", tc.RenterCost, renter.Sub(tax))
+			} else if !host.Equals(tc.HostCost) {
+				t.Errorf("expected host cost %v, got %v", tc.HostCost, host)
+			}
```

Comment:
Since you already went through all the trouble to get this test to use `RenewContract`, you might as well check that the new contract's outputs match `renter + host`. Same for refresh.